openbas-front/src/admin/components/common/injects/InjectDefinition.js

```diff
@@ -349,12 +350,12 @@ class InjectDefinition extends Component {
     super(props);
     this.state = {
       allTeams: props.inject.inject_all_teams,
-      teamsIds: props.inject.inject_teams,
-      assetIds: props.inject.inject_assets,
-      assetGroupIds: props.inject.inject_asset_groups,
+      teamsIds: props.inject.inject_teams || [],
```

Comment:
Nitpick : Shouldn't we make sure that arrays sent back from the backend are not null instead of doing this ?

Reply:
This is not a back-end problem but a store level problem.
When the component is created, we are not sure that the inject object is still fully populated and therefore some properties may be undefined.

Hence the code in the componentDidUpdate method

We can discuss it in person to brainstorm on this type of problem.